parsl/monitoring/db_local.py

```diff
@@ -0,0 +1,218 @@
+                        con.execute(up)
+                        return
+                    if 'task_time_completed' in info.keys() and info['task_time_completed'] is not None:
+                        workflow = self.meta.tables['task']
+                        up = workflow.update().values(task_time_completed=info['task_time_completed']).where(workflow.c.task_id == info['task_id'])\
+                                     .where(workflow.c.task_run_id == run_id)
+                        con.execute(up)
+
+                    # create workflows table if this is a new database without one
+                    if 'workflows' not in self.meta.tables.keys():
+                        workflows = create_workflows_table(self.meta)
+                        self.meta.create_all(con)
+                    # if this is the first sight of the workflow, add it to the workflows table
+                    if len(con.execute(self.meta.tables['workflows'].select(self.meta.tables['workflows'].c.task_run_id == run_id)).fetchall()) == 0:
+                        workflows = self.meta.tables['workflows']
+                        ins = workflows.insert().values(**{k: v for k, v in info.items() if k in workflows.c})
+                        con.execute(ins)
+
+                    # if log has task counts, update the workflow entry in the workflows table
+                    if 'tasks_completed_count' in info.keys():
+                        workflows = self.meta.tables['workflows']
+                        up = workflows.update().values(tasks_completed_count=info['tasks_completed_count']).where(workflows.c.task_run_id == run_id)
+                        con.execute(up)
+                    if 'tasks_failed_count' in info.keys():
+                        workflows = self.meta.tables['workflows']
+                        up = workflows.update().values(tasks_failed_count=info['tasks_failed_count']).where(workflows.c.task_run_id == run_id)
+                        con.execute(up)
+
+                    # create workflow table if this is a new run without one
+                    if 'task' not in self.meta.tables.keys():
+                        workflow = create_workflow_table(self.meta)
+                        self.meta.create_all(con)
+
+                    # check to make sure it is a task log and not just a workflow overview log
+                    if info.get('task_id', None) is not None:
+                        # if this is the first sight of the task in the workflow, add it to the workflow table
+                        if len(con.execute(self.meta.tables['task'].select(self.meta.tables['task'].c.task_id == info['task_id'])
+                                               .where(self.meta.tables['task'].c.task_run_id == run_id)).fetchall()) == 0:
+                            if 'psutil_process_pid' in info.keys():
+                                # this is the race condition that a resource log is before a status log so ignore this resource update
```

Comment:
So specifically here, I'm losing a log line of resource usage information, and hoping that next time round, the task row will have been created?

Reply:
Yes, this drops a resource a log line. I wouldn't expect this to be often unless resource_loop_sleep_duration is very small. I do not know of a more practical response to this situation.